### PR TITLE
Refactor reduce dimension and apply

### PR DESCRIPTION
**Pull request by dgostencnik**
https://git.sinergise.com/team-6/openeo-platform/-/issues/118

### CODE REVIEW

File: src/pg_to_evalscript/javascript_processes/reduce_dimension.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  validateParameter({
+    processName: "reduce_dimension",
+    parameterName: "data",
+    value: data,
+    nullable: false,
+    required: true,
+  });
+
+  validateParameter({
```
Comment (mitjak) [inline src/pg_to_evalscript/javascript_processes/reduce_dimension.js:12]: Dimension can also be validated against type as it should be a `string`.

File: src/pg_to_evalscript/javascript_processes/reduce_dimension.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
+function reduce_dimension(arguments) {
+  const { data, dimension, reducer } = arguments;
```
Comment (mitjak) [inline src/pg_to_evalscript/javascript_processes/reduce_dimension.js:2]: Argument `context` could be passed to `reducer` function, same as above.

**Review by mitjak — CHANGES_REQUESTED** [review]

**Review by mitjak — CHANGES_REQUESTED** [review]

File: src/pg_to_evalscript/javascript_processes/reduce_dimension.js (hunk 1 above)
Comment (dgostencnik) [reply src/pg_to_evalscript/javascript_processes/reduce_dimension.js:12]: was added

File: src/pg_to_evalscript/javascript_processes/apply.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+function apply(arguments) {
+  const { data, process } = arguments;
```
[the remark this answers is not in the document]
Comment (dgostencnik) [reply src/pg_to_evalscript/javascript_processes/apply.js]: Was added in both, `apply` and `reduce_dimension`, but it is probably missing also in some other processes. I don't think context is  used in any predefined process.

**Review by mitjak — APPROVED** [review]

**Pull request merged**